zio-http/src/main/scala/zhttp/service/Handler.scala

```diff
@@ -154,26 +196,65 @@ private[zhttp] final case class Handler[R](
         if (self.isWebSocket(res)) {
           self.upgradeToWebSocket(ctx, jReq, res)
         } else {
-          unsafeWriteAnyResponse(res)
-
           res.data match {
             case HttpData.Empty =>
+              val size = res.data.unsafeSize
+              unsafeWriteAnyResponse(
+                res.updateHeaders(
+                  _ ++
+                    Headers(Name.ContentLength -> size.toString).when(size >= 0)
+                    ++ Headers(Name.TransferEncoding -> Value.Chunked).when(res.data.isChunked),
+                ),
```

Comment:
Code duplication

Reply:
Done, added another method for update response headers